### PR TITLE
Fix copy-pasted typo for read_string return value

### PR DESCRIPTION
**Pull request by shepmaster**

### CODE REVIEW

File: src/libstd/io/mod.rs [hunk 1]
```diff
@@ -1458,7 +1458,7 @@ pub trait BufRead: Read {
     ///
     /// If successful, this function will return the total number of bytes read.
     ///
-    /// An empty buffer returned indicates that the stream has reached EOF.
```
Comment (Centril) [inline src/libstd/io/mod.rs]: For clarity, can we say "`Ok(0)` returned"?

Comment (QuietMisdreavus) [reply src/libstd/io/mod.rs]: How about:

```markdown
If this function returns `Ok(0)`, the stream has reached EOF.
```

Comment (Centril) [reply src/libstd/io/mod.rs]: Sounds great, :shipit:.

**Pull request merged**